src/new/wallet.js

```diff
@@ -1169,12 +1172,9 @@ class HathorWallet extends EventEmitter {
       }
     }
 
-    const tokensBalance = {};
+    const tokensBalance = this._getBalanceRaw();
     for (const tokenUid of Object.keys(tokensHistory)) {
-      const totalBalance = this._getBalanceRaw(tokenUid);
-      totalBalance['transactions'] = transactionCountByToken[tokenUid];
-      // update token total balance
-      tokensBalance[tokenUid] = totalBalance;
+      tokensBalance[tokenUid].transactions = transactionCountByToken[tokenUid];
```

Comment:
I think this might lead to a bug. If we have a token with zero balance, it won't return in the `calculateTokenBalances` method and this will try to access the object.

Actually we must return the token in this balance object because we must show that this token exists, has 0 balance and has X transactions.

We could only check this inside this `for` and create an empty balance for the tokens that are not in the `tokensBalance` object yet.

Reply:
Done